src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -1243,35 +1233,27 @@ private void Uninitialized(object message)
 
         private void TryingToJoin(object message, Address joinWith, Deadline deadline)
         {
-            if (_logInfo && !(message is InternalClusterAction.ITick))
-            {
-                _log.Debug("[TryingToJoin] Received {0}", message);
-            }
-
-            if (message is InternalClusterAction.Welcome)
+            if (message is InternalClusterAction.Welcome w)
             {
-                var w = message as InternalClusterAction.Welcome;
                 Welcome(joinWith, w.From, w.Gossip);
             }
             else if (message is InternalClusterAction.InitJoin)
             {
+                _cluster.LogInfo("Received InitJoin message from [{0}], but this node is not initialized yet", Sender);
                 Sender.Tell(new InternalClusterAction.InitJoinNack(_cluster.SelfAddress));
             }
-            else if (message is ClusterUserAction.JoinTo)
+            else if (message is ClusterUserAction.JoinTo jt)
             {
-                var jt = message as ClusterUserAction.JoinTo;
                 BecomeUninitialized();
                 Join(jt.Address);
             }
-            else if (message is InternalClusterAction.JoinSeedNodes)
+            else if (message is InternalClusterAction.JoinSeedNodes js)
             {
-                var js = message as InternalClusterAction.JoinSeedNodes;
                 BecomeUninitialized();
                 JoinSeedNodes(js.SeedNodes);
             }
-            else if (message is InternalClusterAction.ISubscriptionMessage)
+            else if (message is InternalClusterAction.ISubscriptionMessage isub)
```

Comment:
Implementing C# 7 pattern matching as I pass through these classes, mostly for stylistic reasons and brevity.